app.ts

```diff
@@ -0,0 +1,312 @@
+
+        timeout = setTimeout(() => {
+            func();
+        }, delay);
+    };
+};
+
+function createNavigation() {
+    let recordNav: HTMLElement | null = document.getElementById("record-navigation-container"); // Navigation area
+    if (recordNav !== null) {
+        recordNav.innerHTML = `
+  <div id="navigation-btns">
+    <button value="first" id="first-page-btn">First Page</button>  
+    <button value="previous" id="previous-records-btn">Previous</button>
+    <button value="next" id="next-records-btn">Next</button>
+    <button value="last" id="last-page-btn">Last Page</button>  
+    <button id="confirmation-btn">Get Record</button>
+  </div>
+  <div class="current-page-container">
+    <p id=current-page></p>
+  </div>
+  `;
+    }
+}
+
+function getRecords(fromNumber: number, toNumber: number) {
+    return fetch(`http://localhost:2050/records?from=${fromNumber}&to=${toNumber}`, {
+        method: "GET",
+        headers: { "Content-Type": "application/json" },
+    })
+        .then(checkResponseError)
+        .then((response: Response) => response.text())
+        .then((data) => {
+            let columnDataList = JSON.parse(data);
+            let infoColumns: HTMLElement | null = document.getElementById("info-columns-container"); // Information
+
+            console.log("Hello");
+            (infoColumns as HTMLDivElement).innerHTML = "";
+            for (let i = 0; i < columnDataList.length; i++) {
+                dynamicGrid(columnDataList[i]);
+            }
+
+            let currentPage: HTMLElement | null = document.getElementById("current-page");
+            (currentPage as HTMLParagraphElement).innerHTML = `${fromNumber}  / ${toNumber}.`;
+        })
+        .catch((error: Error) => {
+            console.log(error);
+        });
+}
+
+function recordSelection() {
+    let recordNav: HTMLElement | null = document.getElementById("record-navigation-container"); // Navigation area
+    if (recordNav !== null) {
+        let singleRecordSelection = `
+  <button id="return-btn">Return</button>
+    <div id="user-input-data">
+      <div class="navigation-input-area-id" id="id">
+        <label class="record-labels" for="record-id"
+          >Enter record ID :
+        </label>
+        <input
+          type="text"
+          name="record-id"
+          id="record-id"
+          class="navigation-input"
+          value="0"
+        />
+      </div>
+      <p class="amount-of-records"></p>
+    </div>
+  <button id="get-record-btn">See Record</button>
+  `;
+
+        recordNav.innerHTML = singleRecordSelection;
+
+        let returnBtn: HTMLElement | null = document.getElementById("return-btn");
+        let recordIdInput: HTMLElement | null = document.getElementById("record-id");
+        let numberOfRows = Math.floor(window.innerHeight / 50);
+        let getSingleRecord: HTMLElement | null = document.getElementById("get-record-btn");
+
+        returnBtn?.addEventListener("click", () => {
```

Comment:
The `?` is nice for when you want to do optional things that don't need to be there. But seeing as that you just made the dom elements, you definitly expect them to be there, so you should check if any of them are `null` and handle the errors accordingly.